src/Form/UserPickupRequestForm.php

```diff
@@ -0,0 +1,413 @@
+        $patron_name = $form_state->getValue('patronname');
+        $patron_email = $form_state->getValue('email');
+        $patron_phone = $form_state->getValue('phone');
+        $branch = $form_state->getValue('branch');
+
+        // pickup point/location tied in with time slot
+        $pickup_timeslot = explode('-', $form_state->getValue('pickup_type'));
+
+        $pickupLocationsForRequest = $form_state->get('pickupLocationsForRequest');
+        $locationId_timeslot = explode('-', $form_state->getValue('pickup_type'));
+
+        $selected_titles = array_filter($table_values);
+
+        $holds = [];
+        foreach ($selected_titles as $key=>$val) {
+            array_push($holds, $locker_items[$val]);
+        }
+
+        if (count($holds) == 0) {
+            $messenger->addError(t("There are no request items selected."));
+        } else {  // got at least one hold to be processed
+            $db = \Drupal::database();
+            $guzzle = \Drupal::httpClient();
+            $api_key = \Drupal::config('arborcat.settings')->get('api_key');
+            $api_url = \Drupal::config('arborcat.settings')->get('api_url');
+            $selfCheckApi_key = \Drupal::config('arborcat.settings')->get('selfcheck_key');
+            foreach ($holds as $holdToRequest) {
+                // set the expire date for each selected hold
+                // commented out for now during testing
+                // $updated_hold = $guzzle->get("$api_url/patron/$selfCheckApi_key-$patron_barcode/updated_hold/" . $hold['holdId'] . "?shelf_expire_time=$pickup_date 23:59:59")->getBody()->getContents();
+                // create arborcat_patron_pickup_request records for each of the selected holds
+                $db->insert('arborcat_patron_pickup_request')
+                    ->fields([
+                      'requestId' => $holdToRequest['holdId'],
+                      'patronId' => $pnum,
+                      'branch' => (int) $branch,
+                      'timeSlot' => $locationId_timeslot[1],
+                      'pickupLocation' => $locationId_timeslot[0],
+                      'pickupDate' => $pickup_date,
+                      'contactEmail' => ($notification_types['email'] ? $patron_email : null),
+                      'contactSMS' => ($notification_types['sms'] ? $patron_phone : null),
+                      'contactPhone' => ($notification_types['phone'] ? $patron_phone : null),
+                    ])
+                    ->execute();
+            }
+        }
+
+        // Need to add a c"confirm the request" modal dialog here before proceeding
+
+        $user = \Drupal\user\Entity\User::load(\Drupal::currentUser()->id());
+        $uid = $user->id();
+        $url = \Drupal\Core\Url::fromRoute('entity.user.canonical', ['user'=>$user->id()]);
+
+        return $form_state->setRedirectUrl($url);
+    }
+
+    public function arborcat_mail($key, $email_to, $patron, $code, $holds)
```

Comment:
this looks like a hook function that should go in the .module file. Does it ever get called?